runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterPlugin.java

```diff
@@ -0,0 +1,209 @@
+{
+	private static final Splitter NEWLINE_SPLITTER = Splitter
+		.on("\n")
+		.omitEmptyStrings()
+		.trimResults();
+
+	private static final String CENSOR_MESSAGE = "Hey, everyone, I just tried to say something very silly!";
+
+	private final List<Pattern> filteredPatterns = new ArrayList<>();
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private ChatFilterConfig config;
+
+	@Provides
+	ChatFilterConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(ChatFilterConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		updateFilteredPatterns();
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		filteredPatterns.clear();
+	}
+
+	@Subscribe
+	public void onScriptCallbackEvent(ScriptCallbackEvent event)
+	{
+		if (!"chatFilterCheck".equals(event.getEventName()))
+		{
+			return;
+		}
+
+		int[] intStack = client.getIntStack();
+		int intStackSize = client.getIntStackSize();
+		ChatMessageType chatMessageType = ChatMessageType.of(intStack[intStackSize - 1]);
+
+		// Only filter public chat and private messages
+		switch (chatMessageType)
+		{
+			case PUBLIC:
+			case PUBLIC_MOD:
+			case AUTOCHAT:
+			case PRIVATE_MESSAGE_RECEIVED:
+			case PRIVATE_MESSAGE_RECEIVED_MOD:
+			case CLANCHAT:
+				break;
+			default:
+				return;
+		}
+
+		String[] stringStack = client.getStringStack();
+		int stringStackSize = client.getStringStackSize();
+
+		String message = stringStack[stringStackSize - 1];
+		String censoredMessage = censorMessage(message);
+
+		if (censoredMessage == null)
+		{
+			// Block the message
+			intStack[intStackSize - 2] = 0;
+		}
+		else
+		{
+			// Replace the message
+			stringStack[stringStackSize - 1] = censoredMessage;
+		}
+	}
+
+	@Subscribe
+	public void onOverheadTextChanged(OverheadTextChanged event)
+	{
+		if (!(event.getActor() instanceof Player))
+		{
+			return;
+		}
+
+		String message = censorMessage(event.getOverheadText());
+
+		event.getActor().setOverheadText(message);
+	}
+
+	String censorMessage(String message)
+	{
+		for (Pattern pattern : filteredPatterns)
+		{
+			Matcher m = pattern.matcher(message);
+
+			StringBuffer sb = new StringBuffer();
+
+			while (m.find())
+			{
+				switch (config.filterType())
+				{
+					case CENSOR_WORDS:
+						m.appendReplacement(sb, StringUtils.repeat("*", m.group(0).length()));
+						break;
+					case CENSOR_MESSAGE:
+						return CENSOR_MESSAGE;
+					case REMOVE_MESSAGE:
+						return null;
+				}
+			}
+			m.appendTail(sb);
+
+			message = sb.toString();
+		}
+
+		return message;
+	}
+
+	void updateFilteredPatterns()
+	{
+		filteredPatterns.clear();
+
+		Text.fromCSV(config.censoredWords()).stream()
+			.map(s -> Pattern.compile(Pattern.quote(s), Pattern.CASE_INSENSITIVE))
+			.forEach(filteredPatterns::add);
+
+		NEWLINE_SPLITTER.splitToList(config.censoredRegex()).stream()
+			.map(s ->
+			{
+				try
+				{
+					return Pattern.compile(s, Pattern.CASE_INSENSITIVE);
```

Comment:
I dont think this should be case-insensitive

Reply:
You can't control the case of messages ingame so this should be insensitive